python/dolfinx/fem/linearproblem.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2020 Jørgen S. Dokken
+#
+# This file is part of DOLFINX (https://www.fenicsproject.org)
+#
+# SPDX-License-Identifier:    LGPL-3.0-or-later
+import typing
+
+from petsc4py import PETSc
+from ufl import form
+from dolfinx import fem
+
+
+class LinearProblem():
```

Comment:
I'd suggest adding some methods to access the solver, solution, RHS, etc.

Reply:
I think @jorgensd meant this as a direct access to member variables of the class, `self.solver`, ...

Reply:
I can add getters if that is something we want. As @michalhabera I figured one could just access the obects as problem.solver, problem.u, problem.L etc. as they are public.

Reply:
We just need to provent following:
```python
prob = LinearProblem(a, L)
prob.L = L2
prob.solve()
```
which would incorrectly lead to the impression one can change the form after LinearProblem construction.

Simplest way would be to store such variables into private names `self._L, self._b` and provide a getter for it.

Reply:
Sure, I'll add that